contracts/utils/Exec.sol

```diff
@@ -40,11 +40,14 @@ library Exec {
     }
 
     // get returned data from last call or calldelegate
+    // maxLen - maximum length of data to return, or zero, for the full length
     function getReturnData(uint256 maxLen) internal pure returns (bytes memory returnData) {
         assembly ("memory-safe") {
             let len := returndatasize()
-            if gt(len, maxLen) {
-                len := maxLen
+            if iszero(iszero(maxLen)) {
```

Comment:
1. I believe you mean `not(iszero(maxLen))` here. If so, it is concerning that this function is not covered by a test.
2. You don't need 0 as a special value, `getReturnData` is an `internal pure` function and it costs nothing to pass `type(uint256).max` as a `maxLen` parameter, which is also more explicit.

Reply:
3. And anyways, why did you decide to leave this return data unlimited and not use `REVERT_REASON_MAX_LEN` ?

Reply:
"not" is bitwise, so not(1) is 0xfffffe ... iszero(iszero)) means "nonzero"
blocking maxlen is used during validation, where we want to prevent griefing. during execution, we try to put as few limitations as possible.
e.g. CCIP is a mechanism that returns (via revert) information to client to perform off-chain validation, before making a callback with the result. there is no hard limit on the data size, and we shouldn't arbitrarily put any.